### ISSUE TITLE
[LAB] Week 1_Problem 1_임현진

### ISSUE THREAD

**Issue by HyeonJinIm** [vlab-kaist/NN101_23S#47]
### Problem

Week 1_Problem 1

### Source Code

```python
import torch
from random import random
from typing import Callable
import math

##                        Problem 1                           ##
##                                                            ##
##         Arbitrary quadratic function will be given.        ##
## Return the optimal point(global minimum) of given function ##
##          Condition: highest order term is positive         ##
##                  Made by @jangyoujin0917                   ##
##                                                            ##


def solution(func: Callable, start_point: float):  # DO NOT MODIFY FUNCTION NAME
    lr1 = 1e-4 #알파
    lr2 = 0.9 #베타
    velocity = torch.tensor([1.])
    velocity_0 = velocity.clone().detach()
    x = torch.tensor([start_point], requires_grad=True)
    x_0 = x.clone().detach()
    i = 0
    criteria = 1e4
    #print(x_0.data.item(), velocity_0.data.item())#
    for j in range(10000):
        print(lr1)
        lr1_0 = lr1
        n = torch.tensor([x.data + lr2 * velocity.data], requires_grad=True)
        y = func(n)
        y.backward()
        velocity.data = lr2 * velocity.data - lr1 * n.grad
        x.data += velocity.data
        if abs(velocity.data) < 1e-12:
            #print('Finished at step', i + 1, ': (', x.data.item(), '), diff: ', n.grad.item())#
            break
        #print('Step', i + 1, ":", x.data.item())#
        #print('diff: ', n.grad.item())#
        #print('velocity: ', velocity.data.item())#
        if i == 0 and abs(n.grad.item()) > criteria:
            criteria = abs(n.grad.item())
        i += 1
        if (n.grad > criteria or n.grad < -1 * criteria) or math.isinf(n.grad) or math.isnan(n.grad):
            lr1 = lr1_0 * 1e-1
            velocity = velocity_0.clone().detach()
            x = x_0.clone().detach()
            i = 0
    return x.data.item()

if __name__ == "__main__":
    def test_func(x):  # function for testing;function for evaluation will be different.
        return x ** 8


    t = torch.tensor([10 * random()], requires_grad=True, dtype=torch.float32)
    solution(test_func, t)
```

### Description

I tried to update my learning rate. Rule : If something goes to infinity, make learning rate more smaller and try again.
My copy method of storing x_0 and velocity_0 works well now. I checked it by print function.
But I don't know why it does not converge.
Is my learning rate weird even now?

**Comment by Dongyeongkim** [comment]
This issue is closed now because of the lack of progress
